code/game/machinery/bank_machine.dm

```diff
@@ -95,14 +89,61 @@
 	switch(action)
 		if("siphon")
 			say("Siphon of station credits has begun!")
-			siphoning = TRUE
+			start_siphon()
 			. = TRUE
 		if("halt")
 			say("Station credit withdrawal halted.")
-			end_syphon()
+			end_siphon()
 			. = TRUE
 
-/obj/machinery/computer/bank_machine/proc/end_syphon()
+/obj/machinery/computer/bank_machine/proc/start_siphon()
+	siphoning = TRUE
```

Comment:
```suggestion
	START_PROCESSING(subsystem_type, src)
```

Reply:
eh, siphoning var is used in varioud parts...

Reply:
siphoning var is used only in starting/stopping siphoning and sending siphon status to UI

Reply:
but it can stay if necessary. just don't want process() running just for it to early return

Reply:
ah that makes sense. then all siphon var usage cases can be replaced into that.